aopy/precondition/eye.py

```diff
@@ -0,0 +1,180 @@
+    of potential overshoot components as saccadic events. Default intersaccade 
+    minimum from Sinn & Engbert, 2016 https://doi.org/10.1016/j.visres.2015.05.012 
+    
+    Example:
+
+        Detecting saccades on the first 5 seconds of test data from beignet block 5974 (included in /tests/data/beignet)
+    
+        .. image:: _images/detect_saccades.png
+        .. image:: _images/detect_saccades_hist.png
+        .. image:: _images/detect_saccades_scatter.png
+    
+    Args:
+        eye_kinematics (nt, nch): eye kinematics data, e.g. velocity or acceleration for one eye
+        samplerate (float): sampling rate of the eye data
+        num_sd (float, optional): number of standard deviations above zero to threshold acceleration
+        intersaccade_min (float, optional): minimum time (in seconds) allowed between saccades (from offset to next onset)
+        min_saccade_duration (float, optional): minimum time (in seconds) that a saccade can take (inclusive)
+        max_saccade_duration (float, optional): maximum time (in seconds) that a saccade can take (exclusive)
+        debug (bool, optional): if True, display a figure showing the threshold crossings
+        debug_window (tuple, optional): (start, end) time (in seconds) for the debug figure
+        
+    Returns:
+        tuple: tuple containing:
+        | **onset (nsaccade):** onset time (in seconds) of each detected saccade
+        | **duration (nsaccade):** duration (in seconds) of each detected saccade
+        | **distance (nsaccade):** distance (same units as eye_pos) of each detected saccade
+
+    '''
+    assert thr is None or (len(thr) == 2 and thr[0] > thr[1]), "Threshold must be in the form" 
+    " (positive thr, negative thr)"
+    assert (intersaccade_min is None) or intersaccade_min < max_saccade_duration, "Max saccade"
+    " duration must be longer than the minimum intersaccade interval"
+    if samplerate != 100:
+        print("Warning: this function works best with eye data that has been low-pass filtered below 30 Hz")
+
+    eye_accel = convert_pos_to_accel(eye_pos, samplerate)
+
+    # Set an appropritate threshold to detect saccades
+    if thr is None:
+        baseline_mean = np.mean(eye_accel)
+        baseline_std = np.std(eye_accel)
+        thr = np.mean(baseline_mean) + num_sd*baseline_std
+        thr = (thr, -thr)
+    saccade_onset_time, _ = detect_edges(eye_accel > thr[0], samplerate, rising=True, falling=False)
+    saccade_offset_time, _ = detect_edges(eye_accel < thr[1], samplerate, rising=False, falling=True)
+
+    if saccade_onset_time.size == 0 or saccade_offset_time.size == 0:
+        return [], [], []
+
+    # Only consider saccades that have a threshold-crossing offset
+    onset = []
+    duration = []
+    distance = []
+    prev_offset = 0
+    for t in saccade_onset_time:
+        if intersaccade_min and t < prev_offset + intersaccade_min:
+            continue # ignore saccades within intersaccade_min of the last saccade
```

Comment:
intersaccade_min doesn't have any roles?

Reply:
Yeah by default, saccades can be right next to each other. I found in some data that there are double saccades and if you ignore one of them you miss a big part of the data. But if you set intersaccade_min as 20ms for instance then it will only count the first one in the double.